src/PlaneWaveBasis.jl

```diff
@@ -261,25 +261,76 @@ function PlaneWaveBasis(model::Model{T}, Ecut::Real, fft_size::Tuple{Int, Int, I
-to ensure a minimal k-point density (a certain maximal spacing between k-points).
-
-By default the size of the FFT grid is automatically chosen from the kinetic energy
-cutoff for the Bloch waves and a `supersampling` of 2.0 for the density. This is equal
-to a density cutoff of ``\text{supersampling}^2 * \text{Ecut}`` or `4 * Ecut`.
-A fully custom FFT grid size can also be chosen by specifying the `fft_size` parameter.
-Note, this disables certain symmetry features.
+Creates a plane-wave discretization of a given [`Model`](@ref).
+
+`PlaneWaveBasis` defines the reciprocal-space basis used for Bloch orbitals and the
+real-space FFT grid used for densities and potentials. It bundles the kinetic-energy
+cutoff `Ecut`, k-point sampling into a single object which
+is then used by the Hamiltonian and SCF machinery.
+
+The constructor attempts to choose sensible defaults from the `model` (pseudopotentials,
+recommended cutoffs and symmetries) but allows explicit control of numerical
+parameters through keyword arguments described below.
+
+See [Performing a convergence study](@ref) for some practical
+recommendations how to tune these numerical discretization parameters.
+
+Arguments
+- `model::Model`: Physical model to be discretized. See [`Model`](@ref).
+
+Keyword arguments
+- `Ecut::Number` (default: `recommended_cutoff(model).Ecut`): Kinetic-energy cutoff
+    for Bloch orbitals (units of energy in atomic units). If DFTK cannot infer a
+    recommended cutoff from the pseudopotentials this argument must be supplied.
+- `supersampling::Real` (default: `recommended_cutoff(model).supersampling`): Factor
+    determining the density/potential cutoff relative to `Ecut`. The density cutoff is
+    `supersampling^2 * Ecut` (default supersampling = 2.0, i.e. density cutoff = 4 * Ecut).
+- `kgrid` (default: `KgridSpacing(2π * 0.022)`): Specifies Brillouin-zone sampling.
+    Accepts a [`MonkhorstPack`](@ref) object, a vector of three integers,
+    a [`KgridSpacing`](@ref) object, or any object accepted by `build_kgrid`.
+    The default corresponds to a heuristic spacing.
+- `kshift` (deprecated): Legacy k-point shift. Prefer `kgrid=MonkhorstPack(kgrid, kshift)`.
+- `variational::Bool` (default: `true`): Whether the k-point-specific orbital basis is
+    variationally consistent with the density/potential grid. Non-variational mode is
+    experimental and may disable features.
+- `fft_size::Union{Nothing,Tuple{Int,Int,Int}}` (default: `nothing`): Explicit FFT grid
+    size. If `nothing`, an automatic FFT grid compatible with `Ecut` and `supersampling`
+    is computed. Specifying `fft_size` may disable certain symmetry optimizations.
+- `symmetries_respect_rgrid::Bool` (default: `isnothing(fft_size)`): When `true` the
+    automatically chosen FFT grid will be adjusted to be compatible with common fractional
+    translation denominators (2,3,4,6) to preserve symmetries; set to `false` to skip
```

Comment:
Also not exactly correct. Again I would vouch to be more vague as people can look at the code for details.